### ISSUE TITLE
Feature request: sign releases with PGP

### ISSUE THREAD

**Issue by mitchellrj** [gruntwork-io/terragrunt#1681]
It would be nice to be able to verify release file signatures with PGP, either through signatures of the file directly or signatures of the checksum file. Particularly in light of the recent issues with codecov.

Thank you for your consideration.

**Comment by brikis98** [comment]
Agreed. Not sure when we'll get to it though. FWIW, We do post checksums with the releases already, though I understand that's not the same as signing.

**Comment by mitchellrj** [comment]
That's right. Checksums only validate that what you've downloaded is the same as what was uploaded. The signature helps prove that the uploader also had access to some secret information in addition to the credentials to perform the upload in the first place.

To use a concrete example, the [codecov bash uploader security incident](https://about.codecov.io/security-update/) may have been preventable if they had signed releases of the script and cautious users checked the signatures of those releases.

**Comment by yhakbar** [comment]
We're going to get to this in the relatively near future. Please have patience with us.

**Comment by yhakbar** [comment]
This has been supported for a while now. Closing out issue.
